warehouse/query-core/src/main/java/datawave/query/transformer/GroupingTransform.java

```diff
@@ -362,10 +362,31 @@ private void flatten(List<Document> documents) {
             if (this.groupFieldsSet.contains(shorterName)) {
                 expandedGroupFieldsList.add(shortName);
                 log.trace("{} contains {}", this.groupFieldsSet, shorterName);
-                GroupingTypeAttribute<?> created = makeGroupingTypeAttribute(shortName, field.getData());
-                created.setColumnVisibility(field.getColumnVisibility());
-                fieldMap.put(fieldName, created);
-                fieldToFieldWithContextMap.put(shortName, fieldName);
+                
+                if (field.getData() instanceof Collection<?>) {
+                    // This handles multi-valued entries that do not have gouping context
+                    // Create GroupingTypeAttribute and put in ordered map ordered on the attribute type
+                    Multimap<Type<?>,GroupingTypeAttribute<?>> attrSortedMap = TreeMultimap.create();
```

Comment:
Since we're relying on sorting for correctness, we should declare this as a `SortedMultimap<Type<?>,GroupingTypeAttribute<?>>` to prevent someone from changing this to a HashMultimap for instance.

Reply:
Will change it.